src/Wyam/Program.cs

```diff
@@ -394,6 +396,10 @@ private Engine GetEngine()
                 {
                     engine.CleanOutputPathOnExecute = false;
                 }
+                if (_engineGlobalMetadata != null)
+                {
+                    engine.GlobalMetadata = _engineGlobalMetadata;
```

Comment:
Engine's global metadata is not read-only, so in previous watch mode iterations this metadata may be changed. Maybe we should clone metadata for each engine instance or make metadata read-only.